keyvi/include/keyvi/dictionary/fsa/internal/memory_map_manager.h

```diff
@@ -164,10 +172,19 @@ class MemoryMapManager final {
   void Write(std::ostream& stream, const size_t end) const {
     if (persisted_) {
       for (size_t i = 0; i < number_of_chunks_; i++) {
-        std::ifstream data_file;
-        data_file.open(GetFilenameForChunk(i).native().c_str(), std::ios::binary);
-        stream << data_file.rdbuf();
-        data_file.close();
+        std::ifstream data_stream(GetFilenameForChunk(i).native().c_str(), std::ios::binary);
+
+        if (!data_stream) {
+          throw memory_map_manager_exception("failed to open data stream");
+        }
+
+        stream << data_stream.rdbuf();
+
+        if (!data_stream) {
+          throw memory_map_manager_exception("failed to write into output stream");
+        }
+
+        data_stream.close();
```

Comment:
associated file will automatically be closed when stream object goes out of the scope